integration_tests/src/main/python/repart_test.py

```diff
@@ -15,15 +15,26 @@
 import pytest
 
 from asserts import assert_gpu_and_cpu_are_equal_collect
+from spark_session import is_before_spark_311
 from data_gen import *
 from marks import ignore_order
 
-@pytest.mark.parametrize('data_gen', all_gen, ids=idfn)
+@pytest.mark.parametrize('data_gen', all_gen + [all_basic_struct_gen, StructGen([['child0', DecimalGen(7, 2)]])], ids=idfn)
 def test_union(data_gen):
     assert_gpu_and_cpu_are_equal_collect(
             lambda spark : binary_op_df(spark, data_gen).union(binary_op_df(spark, data_gen)))
 
-@pytest.mark.parametrize('data_gen', all_gen, ids=idfn)
+@pytest.mark.parametrize('data_gen', all_gen + [all_basic_struct_gen, StructGen([['child0', DecimalGen(7, 2)]])], ids=idfn)
+@pytest.mark.skipif(is_before_spark_311(), reason="This is supported only in Spark 3.1.1+")
+def test_union_by_missing_col_name(data_gen):
+    if (isinstance(data_gen, StructGen)):
```

Comment:
xfail would stick around for all further test case iterations once activated in my experience. Some examples of the [documented](https://docs.pytest.org/en/stable/parametrize.html) way of registering [parameter-specific xfail are e.g here](https://github.com/NVIDIA/spark-rapids/blob/63a2e3df1255f09b550d56103525621cbf6bfe56/integration_tests/src/main/python/sort_test.py#L53)

Reply:
I have updated but the xfail doesn't persist or stick, can you please correct me if I am wrong. I tested it by placing the struct data_gen before the all_gen and it was not `xfail`ing every test after struct. 

Nevertheless, I have made the change to follow the convention.